js/vfx.js

```diff
@@ -180,14 +181,15 @@ Events.TaskLevelChanged.subscribe(function (taskInfo) {
     // Only show animations if the level went up
     if (taskInfo.previousLevel >= taskInfo.nextLevel) return;
 
-    let numberOfParticles = 60;
-    let taskProgressBar = getTaskElement(taskInfo.name).querySelector('.progressBar');
-    if (isVisible(taskProgressBar)) {
+    let numberOfParticles = 10;
+    let taskProgressBar = getTaskElement(taskInfo.name)?.querySelector('.progressBar');
+    if (taskProgressBar != null && isVisible(taskProgressBar)) {
         // Don't spawn particles on elements that are invisible
         ParticleSystem.onetimeSplash(taskProgressBar, numberOfParticles);
         VFX.flash(taskProgressBar);
     }
     let quickTaskProgressBar = document.querySelector(`#quickTaskDisplay .${taskInfo.type}.progressBar`);
+    if (!quickTaskProgressBar) return;
```

Comment:
Das seh ich, weil wir nicht für jeden TaskType unbedingt ne QuickBar haben.
Aber, bitte in JS immer möglichst präzise vergleichen, sprich:
```suggestion
    if (quickTaskProgressBar === null) return;
```
Ansonsten baut man sich alle möglichen Arten von Verhalten in seinen Code, weil JS sehr liberal ist, was alles true bzw false ist.
querySelector und Konsorten geben alle immer `null` zurück, wenn sie nichts finden.